matrix/Scalar.hpp

```diff
@@ -17,21 +17,18 @@ template<typename Type>
 class Scalar
 {
 public:
-    Scalar() : _value()
-    {
-    }
+    Scalar() = default;
```

Comment:
Is this useful?

Reply:
Nope, originally wasn't sure if we could get away with Scalar being constant, but it appears that we can.